evac/rvo2_dto.py

```diff
@@ -92,34 +79,35 @@ def _find_closest_exit(self, e):
                 continue
             if self._next_room_in_smoke(e, path) is not True:
                 try:
-                    paths_free_of_smoke.append([x, y, LineString(path).length, exit])
+                    paths_free_of_smoke.append([x, y, LineString(path).length, exit, path])
                 except:
                     self.evacuees.set_finish_to_agent(e)
                     if exit['type'] == 'teleport':
                         evacuee.target_teleport_coordinates = (exit['direction_x'], exit['direction_y'])
                     if evacuee.target_teleport_coordinates is not None:
                         self.append_agents_to_move_downstairs(evacuee, e)
-                    paths_free_of_smoke.append([x, y, 0, exit])
+                    paths_free_of_smoke.append([x, y, 0, exit, path])
 
-                # paths.append([x, y, LineString(path).length])
             else:
                 paths.append([x, y, LineString(path).length, exit])
```

Comment:
When the only path for agent is covered with smoke this list index  4 is out of range. Possibly the `path` should be added also to this list.

Reply:
I have already corrected a mistake that should not have happened